src/Operation/Find.php

```diff
@@ -293,33 +293,49 @@ public function execute(Server $server)
         return $cursor;
     }
 
+    public function getCommandDocument()
+    {
+        return $this->createCommandDocument();
+    }
+
     /**
-     * Create options for executing the command.
-     *
-     * @see http://php.net/manual/en/mongodb-driver-server.executequery.php
-     * @return array
+     * Construct a command document for Find
      */
-    private function createOptions()
+    private function createCommandDocument()
     {
-        $options = [];
+        $cmd = ['find' => $this->collectionName, 'filter' => new BSONDocument($this->filter)];
 
-        if (isset($this->options['readPreference'])) {
-            $options['readPreference'] = $this->options['readPreference'];
+        $options = $this->createFindOptions();
+
+        if (empty($options)) {
+            return $cmd;
         }
 
-        if (isset($this->options['session'])) {
-            $options['session'] = $this->options['session'];
+        // maxAwaitTimeMS is a Query level option so should not be considered here
+        unset($options['maxAwaitTimeMS']);
+
+        if (isset($options['cursorType'])) {
+            if ($options['cursorType'] === self::TAILABLE) {
+                $cmd['tailable'] = true;
+            }
+            if ($options['cursorType'] === self::TAILABLE_AWAIT) {
+                $cmd['tailable'] = true;
+                $cmd['awaitData'] = true;
+            }
         }
 
-        return $options;
+        return $cmd + $options;
```

Comment:
`modifiers` may need special handling here, since that isn't a `find` option at all.

It may help to look at what Query constructor does:

 * [`hint` option falls back to `$hint` modifier](https://github.com/mongodb/mongo-php-driver/blob/1.4.1/src/MongoDB/Query.c#L138)
 * [Other options fall back to similarly named modifiers](https://github.com/mongodb/mongo-php-driver/blob/1.4.1/src/MongoDB/Query.c#L270) (see macros linked with `else`)

One approach here might be to create a local array of pairs, each containing the original option and modifier fallback. Iterating on this array, you could then check if the modern option is not set and then assign the modifier's value. That is essentially what the `PHONGO_QUERY_OPT_X()` macros in PHPC do.

Normally, I'd suggest using `array_key_exists()` to be correct, but `isset()` should work fine in this case based on the logic in `createQuery()` (now `createFindOptions()`). This operation doesn't assign any options with `null` values. Differentiating `null` values from missing keys is the main reason we'd want to use the more-expensive `array_key_exists()` function.

Reply:
Second thought: since you've already called `createFindOptions()` before this point, any modifiers will be under a `modifiers` option, which will have an array type. Therefore, you can pluck the relevant fallback values from there and finally `unset($options['modifiers'])` after the fact.